### ISSUE TITLE
Modify full_biomass and comb_ files to indicate where litter biomass = 0 vs. NULL

### ISSUE THREAD

**Issue by nutnetadmin** [nutnetadmin/nutnet-data-tracker#239]
A subset of the fix to #50; this one just focused on live and litter biomass.

Here's my plan: 

Update the "notes" column in the "mass" table. 
If a site did not collect litter in that year, we'll put a note in that says "litter not collected."

In the full_biomass view: 
If live is not NULL and if note <> "litter not collected," then full_biomass.dead_mass = 0

Otherwise, dead mass = NULL, as per usual.

Litter not collected sites/years: 
bari.ar 2016

> That year we  hired a person to do it and she made a mistake by not wheighting it. We only realized after she has finished and the samples were discarded so we do not have that measure. She said there wasn't much dead biomass but we cannot assure that.

temple.us, all years

> It’s a some-of-the-above kind of thing.  For all years of the experiment, we have harvested only current year biomass, which is pretty easy for us to distinguish.  Beginning in 2012, each February we cut and removed the previous years standing dead and lightly raked the plots. Thus, from 2012 forward there is no litter. But we have not quantified litter mass in any year (unfortunately).

shps.us, all years

> From 2007-2012, we clipped biomass. We always separated into three classes “Annual”,”Perennial (herbaceous)” and “Shrub” since we have to process each of those a little differently. ONLY in 2007 do I have a different category listed: “Dead.” I think this is standing dead (carryover from previous year), NOT litter. I have a note that it should not be included in biomass calculations. So, you should have NO litter observations from shps.us!

spin.us, hall.us, all years

> The first point I should make is that we are only reporting standing dead material as ‘litter’ at both Spindletop and Hall’s Prairie.

**Comment by nutnetadmin** [comment]
add an additional kind of note: 
"true zero: no biomass"

where there really was no biomass (live, litter or otherwise), let this be zero

In the comb_ view:
If live is NULL and litter is NULL and note = "true zero: no biomass" then full_biomass.dead_mass = 0
If live is NULL and litter is NULL and note = "true zero: no biomass" then full_biomass.live_mass = 0

**Comment by nutnetadmin** [comment]
fixed with fixing-litter-and-live-to-zero.sql

there are now three columns for biomass: live-mass, litter-mass, and standing-dead-mass.

**Comment by nutnetadmin** [comment]
```

#use nutnet_local_2019;
#set @@sql_mode = 'STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION';

select * from site;

select * from full_biomass;


################################################################
# Fix 1: Where live_mass is NULL and dead_mass is not NULL, make live_mass = 0.
# (in comb_ tables, resutling from biomass_plot_summary)
################################################################

# query out observations where live mass is null, but dead mass is not: 
select distinct site_code, year, plot, live_mass, dead_mass, total_mass from comb_by_plot
where (live_mass IS NULL and dead_mass IS NOT NULL);

# put this in biomass_temp. We'll add zero for live mass.
truncate table biomass_temp;
select * from biomass_temp;

INSERT into biomass_temp
(site, plot, date)
select distinct site_code, plot, year from comb_by_plot
where (live_mass is NULL and dead_mass is not NULL);

# ethabuka checks out
select distinct site, date from biomass_temp;
/*
elliot.us	2014
hart.us	2012
kiny.au	2018
mcla.us	2008
ethass.au	2013
ethass.au	2014
ethass.au	2015
ethass.au	2016
ethamc.au	2013
ethamc.au	2015
ethamc.au	2014
ethamc.au	2018
ethamc.au	2019
ethamc.au	2016
*/

# ethabuka & elliot both check out against raw data; these shoudl be zeros.
select * from biomass_temp where site = 'hart.us'; # plot 15 2012
# checked against raw data. live biomass not listed in raw data, assuming this is zero.
select * from biomass_temp where site = 'kiny.au'; # plot 20, 27 2018
# checked against raw data. zeros are listed.
select * from biomass_temp where site = 'mcla.us'; # plot 1, 8 2008
# raw data do not exist in my files (or anywhere??). assume zero.

update biomass_temp set mass = 0;
update biomass_temp set taxa = 'live';
update biomass_temp set note = 'filled in live biomass with zero in 2019';
select * from biomass_temp;

# join to appropriate tables for insert.
select subplot_id, biomass_category_id, mass, note, date
from biomass_temp
left join site on (site.site_code = biomass_temp.site)
left join plot on (site.site_id = plot.site_id and plot.plot = biomass_temp.plot)
left join subplot on (plot.plot_id = subplot.plot_id and subplot.subplot_type_id = 1)
left join biomass_category on (biomass_category.category = biomass_temp.taxa);

# insert these into biomass table
# (first, need to modify biomass table to accomodate notes) -- did this with GUI tools
START TRANSACTION;
insert into biomass
(subplot_id, biomass_category_id, mass, note, year, date)
select subplot_id, biomass_category_id, mass, note, date, left(date, 4)
from biomass_temp
left join site on (site.site_code = biomass_temp.site)
left join plot on (site.site_id = plot.site_id and plot.plot = biomass_temp.plot)
left join subplot on (plot.plot_id = subplot.plot_id and subplot.subplot_type_id = 1)
left join biomass_category on (biomass_category.category = biomass_temp.taxa);

select * from biomass order by biomass_id desc limit 100;

COMMIT;


# check that this worked:
select distinct site_code, year, plot, live_mass, dead_mass, total_mass from comb_by_plot
where (live_mass is NULL and dead_mass is not NULL);
# fixed!

################################################################
# Fix 2: Where dead_mass is NULL and live_mass is not NULL, make dead_mass = 0.
################################################################

# query out these observations
select distinct site_code, year, plot, live_mass, dead_mass, total_mass from comb_by_plot
where (dead_mass is NULL and live_mass is not NULL) limit 10000;
# 2978 rows.

# put this in biomass_temp;
truncate table biomass_temp;
select * from biomass_temp;

INSERT into biomass_temp
(site, plot, date)
select distinct site_code, plot, year from comb_by_plot
where (dead_mass is NULL and live_mass is not NULL);

# correcting for site-level factors (mowing, e.g.): 
select distinct site, count(*) from biomass_temp group by site;
# 57 sites

# result of mowing/burning maybe?
select distinct continent, site_code, managed, burned, grazed, anthropogenic, 
(managed+burned + grazed +anthropogenic) as manage_factors,
count(*) as num_rows
from biomass_temp 
left join site on (site.site_code = biomass_temp.site) 
group by site_code
order by continent, num_rows desc;
# 57 sites

# start with Europe. 
# who mows? 
select distinct continent, site_code, managed, burned, grazed, anthropogenic, 
(managed+burned + grazed +anthropogenic) as manage_factors,
count(*) as num_rows
from biomass_temp 
left join site on (site.site_code = biomass_temp.site)
where continent = 'Europe' 
group by site_code
order by continent, num_rows desc;
# 11 sites
/*
Europe	frue.ch	1	0	1	1	3	208
Europe	bayr.de	1	0	0	1	2	90
Europe	lancaster.uk	1	0	1	0	2	82
Europe	comp.pt	1	0	1	1	3	81
Europe	nioo.nl	0	0	1	1	2	51
Europe	valm.ch	0	0	0	0	0	9
Europe	badlau.de	1	0	0	1	2	6
Europe	burren.ie	1	0	1	0	2	4
Europe	rook.uk	0	0	0	0	0	2
Europe	gall.it	1	0	0	0	1	2
Europe	kilp.fi	0	0	1	0	1	1
*/

################################################################
# Fix 2A: Separating Dead --> standing_dead and true litter
################################################################
/*Four sites report sometimes report “Dead", but it’s really standing dead, and is thus an underestimate of litter 
spin.us
hall.us
shps.us
temple.us
*/

select * from biomass_category;
select distinct site_code from full_biomass where category = 'STANDING DEAD';
# BARTA.US
# CHILCAS.AR
# HALL.US
# KIBBER.IN

select distinct site_code, category, concat(CHAR(year)) from full_biomass where site_code in 
('spin.us',
'hall.us',
'shps.us',
'temple.us')
and live = 0
group by site_code, category;

/*hall.us	LITTER	2007,2008,2009,2010,2011,2014
hall.us	STANDING DEAD	2012,2013
shps.us	LITTER	2007
spin.us	LITTER	2007,2008,2009,2010,2011,2012,2013,2014,2015,2016,2017,2018*/

update biomass
left join subplot using (subplot_id)
left join plot using (plot_id)
left join site using (site_id)
set biomass_category_id = (select biomass_category_id from biomass_category where category = 'STANDING DEAD')
where site_code in 
('spin.us',
'hall.us',
'shps.us',
'temple.us')
and biomass_category_id = (select biomass_category_id from biomass_category where category = 'LITTER');

select distinct site_code, category from full_biomass where site_code in ('spin.us',
'hall.us',
'shps.us',
'temple.us')
and live = 0;

select distinct site_code, year from comb_by_plot  
where standing_dead_mass<>litter_mass;
# ok this is good

select distinct site_code, year from comb_by_plot  
where live_mass is not NULL and (litter_mass is NULL OR standing_dead_mass is NULL);

select site_code, plot, year, live_mass, litter_mass, total_mass, dead_mass from comb_by_plot where site_code = 'sevi.us';
select * from full_biomass where site_code = 'sevi.us' and live = 0;

################################################################
# ...Back to fix # 2 Where dead_mass is NULL and live_mass is not NULL, make dead_mass = 0.
################################################################

# insert some zeros for litter biomass
truncate table biomass_temp;

INSERT into biomass_temp
(site, plot, date)
select distinct site_code, plot, year from comb_by_plot
where (litter_mass is NULL AND standing_dead_mass is NULL) and live_mass is not NULL;
# 2980 rows

update biomass_temp set taxa = 'LITTER';
update biomass_temp set mass = 0;
update biomass_temp set note = 'inserted 0s for litter biomass in 2019';

# some exceptions where this won't work:
delete from biomass_temp where site = 'bari.ar' and date = 2016;
/*That year we  hired a person to do it and she made a mistake by not wheighting it. 
We only realized after she has finished and the samples were discarded so we do not have 
that measure. She said there wasn't much dead biomass bu
t we cannot assure that.
*/

# In Temple, there is no litter, but only starting in 2012. 
/*It’s a some-of-the-above kind of thing. For all years of the experiment, 
we have harvested only current year biomass, which is pretty easy for us to distinguish. 
Beginning in 2012, each February we cut and removed the previous years standing dead and 
lightly raked the plots. Thus, from 2012 forward there is no litter. 
But we have not quantified litter mass in any year (unfortunately).*/
delete from biomass_temp where site = 'temple.us' and date < 2012;

# shps.us - standing dead in 2007; litter not harvested in any other year.
/*From 2007-2012, we clipped biomass.
 We always separated into three classes “Annual”,”Perennial (herbaceous)” and 
 “Shrub” since we have to process each of those a little differently. ONLY in 2007 
 do I have a different category listed: “Dead.” I think this is standing dead 
 (carryover from previous year), NOT litter. I have a note that it should not be 
 included in biomass calculations. So, you should have NO litter observations from shps.us!*/
delete from biomass_temp where site = 'shps.us';
select * from full_biomass where site_code = 'shps.us' and live = 0; # just standing dead.

# spin and hall - litter is NULL; only standing dead.
/*The first point I should make is that we are only 
reporting standing dead material as ‘litter’ at both Spindletop and Hall’s Prairie.*/
delete from biomass_temp where site in ('spin.us', 'hall.us');
select * from full_biomass where site_code in ('spin.us', 'hall.us') and live = 0; # just standing dead.

# still not sure
# kiny, lancaster, ufrec, sevi
delete from biomass_temp where site = 'sevi.us';
delete from biomass_temp where site = 'lancaster.uk';
delete from biomass_temp where site = 'ufrec.us';
delete from biomass_temp where site = 'kiny.au';

# insert the zeros into the database
select count(*) from biomass_temp;
# 1960 rows to insert

START TRANSACTION;
insert into biomass
(subplot_id, biomass_category_id, mass, note, year, date)
select subplot_id, biomass_category_id, mass, note, date, left(date, 4)
from biomass_temp
left join site on (site.site_code = biomass_temp.site)
left join plot on (site.site_id = plot.site_id and plot.plot = biomass_temp.plot)
left join subplot on (plot.plot_id = subplot.plot_id and subplot.subplot_type_id = 1)
left join biomass_category on (biomass_category.category = biomass_temp.taxa);

select * from biomass order by biomass_id desc limit 100;

# check that this worked:
select distinct site_code, year from comb_by_plot
where (live_mass is NOT NULL and dead_mass is NULL);
# fixed!

################################################################
# Fix #3: Plots that truly, honestly, have no biomass
################################################################

SELECT 
distinct site_code, year, 
concat('where site_code = ', "'", site_code, "'", ' and year = ', year) as string,
count(*) as num_plots,
sum(case when live_mass is null and (litter_mass is NULL AND standing_dead_mass is NULL) then 1 else 0 end) AS num_missing,
group_concat(case when live_mass is null and (litter_mass is NULL AND standing_dead_mass is NULL) is NULL then plot else '' end SEPARATOR '/')
FROM comb_by_plot
group by site_code, year
having num_missing >0
and num_missing < num_plots; # this gets rid of cases where there is no biomass data for that site*year at all

# put these into biomass_temp, but only if truly zero, not truly missing
truncate table biomass_temp;
#pich.ec	2013	where site_code = 'pich.ec' and year = 2013	50	20	/37/45/////32/40/48////35/43//////38/46////33/41/49/////36/44/////31/39/47////34/42/50///
# truly missing

#niwo.us	2007	where site_code = 'niwo.us' and year = 2007	34	14	4////11//15///38/3/////30/13////2//40//24///12/33//36/1//
# truly missing

#ethamc.au	2016	where site_code = 'ethamc.au' and year = 2016	30	9	/29///9/////3/26///13/////12//////18/23/30///
#ethamc.au	2014	where site_code = 'ethamc.au' and year = 2014	30	2	////////////23////////////21/////
#ethamc.au	2015	where site_code = 'ethamc.au' and year = 2015	30	2	/11///////////////28/////////////
#ethass.au	2013	where site_code = 'ethass.au' and year = 2013	30	3	/////////6///////////9////19/////
#ethass.au	2015	where site_code = 'ethass.au' and year = 2015	30	1	///////////////7//////////////
#ethass.au	2016	where site_code = 'ethass.au' and year = 2016	30	1	/////////////////////////////24
#ethass.au	2014	where site_code = 'ethass.au' and year = 2014	30	2	//////////////////////23///////6
# truly zero: checked against raw data
insert into biomass_temp (site, plot, date) 
select site_code, plot, year from comb_by_plot 
where site_code in ('ethass.au', 'ethamc.au') and year between 2013 and 2016
AND (live_mass is NULL and (litter_mass is NULL AND standing_dead_mass is NULL))
AND (proportion_par is not NULL OR rich is not null);

# drop the one that is truly missing - plot 6, 2013
delete from biomass_temp where site = 'ethass.au' and date = 2013 and plot = 6;


#frue.ch	2014	where site_code = 'frue.ch' and year = 2014	24	8	////7///17///19/27/29//10//6///26////
# frue.ch 2014 plots 6, 7, 10... — ugh, this is a case of the subplot that was provided in 
# 2014 not matching the core subplot, so it is being masked from the shared database. 
# I’ll just update the subplot, that’s what EL and I have been doing. 
# Added to the list. https://github.com/nutnetadmin/nutnet-data-tracker/issues/242
select site_code, date, subplot, subplot_type_id from biomass 
left join subplot using (subplot_id) left join plot using (plot_id) left join site using (site_id)
where site_code = 'frue.ch' and year = 2014 and subplot_type_id = 3;

START TRANSACTION;
update biomass
left join subplot using (subplot_id) left join plot using (plot_id) left join site using (site_id)
set subplot_id = (select subplot_id from subplot where subplot.plot_id = plot.plot_id and subplot_type_id = 1)
where site_code = 'frue.ch' and year = 2014 and subplot_type_id = 3;
COMMIT;


#summ.za	2012	where site_code = 'summ.za' and year = 2012	30	6	///30/1////16//21/////10//15////////////
#summ.za	2011	where site_code = 'summ.za' and year = 2011	30	3	//////////30/////16///////10///////
# another case of a non-core subplot?
select site_code, date, subplot, subplot_type_id from biomass 
left join subplot using (subplot_id) left join plot using (plot_id) left join site using (site_id)
where site_code = 'summ.za' and year in (2011, 2012);

# probably just nulls -- 30, 16, 10 
select site_code, plot, N, P, K, Exclose, subplot, subplot_type_id from subplot 
left join plot using (plot_id) left join site using (site_id)
where site_code = 'summ.za' and plot in (10, 16, 30, 1, 21, 15);
# extra control plots maybe? also plot 16 has a 'B' subplot?

select site_code, date, subplot, subplot_type_id from biomass 
left join subplot using (subplot_id) left join plot using (plot_id) left join site using (site_id)
where site_code = 'summ.za' and plot =16;
# never sampled past 2010

#marc.ar	2012	where site_code = 'marc.ar' and year = 2012	30	4	////////17//11////////////////8///6
#marc.ar	2013	where site_code = 'marc.ar' and year = 2013	30	4	/11////////////////8///6/////////17
#marc.ar	2014	where site_code = 'marc.ar' and year = 2014	30	4	///////////8///6/////////17//11////
#marc.ar	2015	where site_code = 'marc.ar' and year = 2015	30	4	//6/////////17//11////////////////8
select site_code, plot, N, P, K, Exclose, Treat_other_name, treat_other_level, subplot, subplot_type_id from subplot 
left join plot using (plot_id) left join site using (site_id)
where site_code = 'marc.ar' and plot in (6, 8, 11, 17);
# extra control plots maybe - just nulls

#anti.ec	2013	where site_code = 'anti.ec' and year = 2013	50	2	///////////////60//////////////////////////////32////
# plot 60? but there are only 50 plots. 
select * from comb_by_plot where site_code = 'anti.ec' and plot > 50;
select distinct site_code, block, plot from comb_by_plot where site_code = 'anti.ec';

# oh, update this
START TRANSACTION;
select * from plot left join site using (site_id) where site_code = 'anti.ec';
update plot left join site using (site_id)
set plot = plot-10 
where site_code = 'anti.ec' and plot >40;
COMMIT;

# true zeros in the raw data
insert into biomass_temp (site, plot, date) 
select site_code, plot, year from comb_by_plot 
where site_code in ('anti.ec')
AND (live_mass is NULL and (standing_dead_mass is NULL AND litter_mass is NULL))
AND (proportion_par is not NULL OR rich is not null);


#cbgb.us	2015	where site_code = 'cbgb.us' and year = 2015	54	2	//44////////////15///////////////////////////////////////
#cbgb.us	2018	where site_code = 'cbgb.us' and year = 2018	54	2	////////26/////////////////////22////////////////////////
# wrote an email to ask
# email reply = missing; wasp nests (!)

#konzn1b.us	2007	where site_code = 'konzn1b.us' and year = 2007	12	2	/////////6/9/
# don't have these raw data, assume null
select site_code, plot, N, P, K, Exclose, Treat_other_name, treat_other_level, subplot, subplot_type_id from subplot 
left join plot using (plot_id) left join site using (site_id)
where site_code = 'konzn1b.us' and plot in (6, 9);
# extra control plots maybe

#bari.ar	2015	where site_code = 'bari.ar' and year = 2015	30	1	///////6//////////////////////
# this is supicious: 
# according to raw data these are missing, 
# but it looks like we have too many values for plot 5 on top of it all.
# sent an email to ask
# assume that both plots are wrong

select site_code, date, subplot, subplot_type_id from biomass 
left join subplot using (subplot_id) left join plot using (plot_id) left join site using (site_id)
where site_code = 'bari.ar' and year in (2015) and plot in (5,6);

START TRANSACTION;
delete biomass from biomass
left join subplot using (subplot_id) left join plot using (plot_id) left join site using (site_id)
where site_code = 'bari.ar' and year in (2015) and plot in (5,6);
COMMIT;

#comp.pt	2013	where site_code = 'comp.pt' and year = 2013	31	1	//////////////////////////////14
#comp.pt	2017	where site_code = 'comp.pt' and year = 2017	31	1	////////14//////////////////////
#comp.pt	2018	where site_code = 'comp.pt' and year = 2018	31	1	////////14//////////////////////
#comp.pt	2019	where site_code = 'comp.pt' and year = 2019	31	1	////////////////////////14//////
# plot 14 always -- how come?
select site_code, date, subplot, subplot_type_id from biomass 
left join subplot using (subplot_id) left join plot using (plot_id) left join site using (site_id)
where site_code = 'comp.pt' and plot = 14;
# nope, really missing this plot from years 2012+
# not sampled/extra treatment
# cool! good to go.



#hogone.us	2015	where site_code = 'hogone.us' and year = 2015	20	1	///////////////////17
# 17 is truly lost, checked in raw data

#kbs.us	2016	where site_code = 'kbs.us' and year = 2016	50	1	/////////////15////////////////////////////////////
# "no data, no biomass? Not Harvest in Field."

#kibber.in	2014	where site_code = 'kibber.in' and year = 2014	30	1	///////////////18//////////////
# not recorded in data

#lancaster.uk	2016	where site_code = 'lancaster.uk' and year = 2016	24	1	///////////////////////19
# "Not able to take sample due to massive amount of Juncus effusus (over 1.5m tall across subplot)"

#look.us	2008	where site_code = 'look.us' and year = 2008	30	1	//////////////////////26///////
# data not in folders i can access

#mtca.au	2017	where site_code = 'mtca.au' and year = 2017	40	1	///////////36////////////////////////////
# not recorded, I could assume a zero? 
# sent an email.
# no reply, assume missing

#sgs.us	2015	where site_code = 'sgs.us' and year = 2015	30	1	/////////////6////////////////
#sgs.us	2016	where site_code = 'sgs.us' and year = 2016	30	1	/////6////////////////////////
# there is definitely biomass in plot 6! another case of an incorrect subplot?
select site_code, year, date, subplot, subplot_type_id from biomass 
left join subplot using (subplot_id) left join plot using (plot_id) left join site using (site_id)
where site_code = 'sgs.us' and plot = 6;

# yes, wrong subplot
START TRANSACTION;
update biomass
left join subplot using (subplot_id) left join plot using (plot_id) left join site using (site_id)
set subplot_id = (select subplot_id FROM subplot where subplot.plot_id = plot.plot_id and subplot_type_id = 1)
where site_code = 'sgs.us' and year in (2015, 2016) and subplot_type_id = 3;
COMMIT;


#sier.us	2014	where site_code = 'sier.us' and year = 2014	50	1	/////////////////22////////////////////////////////
select site_code, year, date, subplot, subplot_type_id from biomass 
left join subplot using (subplot_id) left join plot using (plot_id) left join site using (site_id)
where site_code = 'sier.us' and plot = 22;
# yes, wrong subplot
START TRANSACTION;
update biomass
left join subplot using (subplot_id) left join plot using (plot_id) left join site using (site_id)
set subplot_id = (select subplot_id FROM subplot where subplot.plot_id = plot.plot_id and subplot_type_id = 1)
where site_code = 'sier.us' and year in (2014) and subplot_type_id = 3;
COMMIT;


#trel.us	2016	where site_code = 'trel.us' and year = 2016	30	1	//////////////////23///////////
select site_code, year, date, subplot, subplot_type_id from biomass 
left join subplot using (subplot_id) left join plot using (plot_id) left join site using (site_id)
where site_code = 'trel.us' and plot = 23;
# looks like it is truly missing
# not in raw data either
# this is a control plot, and the only one at that site

#yarra.au	2014	where site_code = 'yarra.au' and year = 2014	40	1	/////////////////////////////////////41//
# sent an email

# well I can insert these biomass values anyways!
select * from biomass_temp;
# two from anitsana, the rest from ethabuka


START TRANSACTION;
insert into biomass
(subplot_id, biomass_category_id, 
mass, note, year, date)
select subplot_id, 
(select biomass_category_id from biomass_category where category = 'LIVE'),
0, 'inserted 0s for live+litter biomass in 2019', date, left(date, 4)
from biomass_temp
left join site on (site.site_code = biomass_temp.site)
left join plot on (site.site_id = plot.site_id and plot.plot = biomass_temp.plot)
left join subplot on (plot.plot_id = subplot.plot_id and subplot.subplot_type_id = 1);

insert into biomass
(subplot_id, biomass_category_id, 
mass, note, year, date)
select subplot_id, 
(select biomass_category_id from biomass_category where category = 'LITTER'),
0, 'inserted 0s for live+litter biomass in 2019', date, left(date, 4)
from biomass_temp
left join site on (site.site_code = biomass_temp.site)
left join plot on (site.site_id = plot.site_id and plot.plot = biomass_temp.plot)
left join subplot on (plot.plot_id = subplot.plot_id and subplot.subplot_type_id = 1);

select * from biomass order by biomass_id desc limit 100;

SELECT 
site_code, year, plot, live_mass, litter_mass, standing_dead_mass FROM comb_by_plot
where
site_code in ('anti.ec', 'ethass.au', 'ethamc.au'); # this gets rid of cases where there is no biomass data for that site*year at all


commit;



select * from comb_byplot_clim_soil order by site_code, year, plot  limit 6000000


```
